fhir-smart/src/main/java/com/ibm/fhir/smart/AuthzPolicyEnforcementPersistenceInterceptor.java

```diff
@@ -0,0 +1,268 @@
+                                return;
+                            }
+                        }
+                    } catch (Exception e) {
+                        log.log(Level.WARNING, "Unexpected exception while processing inclusionCriteria '" + searchParm +
+                                "' in the Patient compartment for resource type " + resourceType);
+                    }
+                }
+            } catch (FHIRSearchException e) {
+                log.log(Level.WARNING, "Unexpected exception while enforcing authorization policy in the Patient compartment"
+                        + " for resource type " + resourceType);
+            }
+        }
+
+        if (approvedScopeMap.containsKey(ContextType.USER)) {
+            throw new UnsupportedOperationException("SMART scopes with context type 'user' are not yet supported.");
+        }
+
+
+        String msg = "Requested interaction is not permitted by any of the passed scopes.";
+        throw new FHIRPersistenceInterceptorException(msg)
+                .withIssue(FHIRUtil.buildOperationOutcomeIssue(msg, IssueType.FORBIDDEN));
+    }
+
+    /**
+     * @param node
+     * @return the id to the Patient resource referenced by this node (assuming it is a Reference with a valid
+     *         reference value); otherwise null
+     */
+    private String getPatientRefVal(FHIRPathNode node) {
+        if (!node.isElementNode() || !node.asElementNode().element().is(Reference.class)) {
+            throw new IllegalStateException("Patient compartment inclusionCriteria expression has returned a non-Reference");
+        }
+        Reference reference = node.asElementNode().element().as(Reference.class);
+        if (reference.getReference() != null && reference.getReference().hasValue()) {
+            String refVal = reference.getReference().getValue();
+            if (refVal != null && refVal.startsWith(PATIENT_REF_PREFIX)) {
+                return refVal.substring(PATIENT_REF_PREFIX.length());
+            }
+        } else if (log.isLoggable(Level.FINE)){
+            log.fine("Skipping non-patient / non-relative reference: '" + reference + "'");
+        }
+
+        return null;
+    }
+
+    /**
+     * @param permission
+     * @param requiredPermission
+     * @return true if the grantedPermission includes the requiredPermission; otherwise false
+     */
+    private boolean hasPermission(Permission grantedPermission, Permission requiredPermission) {
+        if (grantedPermission == Permission.ALL) {
+            return true;
+        } else {
+            return grantedPermission == requiredPermission;
+        }
+    }
+
+    private String getAccessToken() throws FHIRPersistenceInterceptorException {
+        List<String> list = FHIRRequestContext.get().getHttpHeaders().get("Authorization");
+        if (list.size() != 1) {
+            throw new FHIRPersistenceInterceptorException("Request must contain exactly one Authorization header.");
```

Comment:
interesting check...

Reply:
just trying to be defensive there, not attached to it.  re-open if you think it should be changed.